packages/pyright-internal/src/analyzer/semanticTokensWalker.ts

```diff
@@ -617,19 +623,32 @@ export class SemanticTokensWalker extends ParseTreeWalker {
                         effectiveType = FunctionType.getEffectiveReturnType(methodType);
                 }
 
-                if (effectiveType && isFunction(effectiveType)) {
-                    const isMethod = effectiveType.shared.declaration?.isMethod;
+                // If there is no type information, use `property`
+                if (!effectiveType) {
+                    return SemanticTokenTypes.property;
+                }
+
+                if (isFunction(effectiveType) || isOverloaded(effectiveType)) {
+                    const isMethod = isMethodType(effectiveType);
                     return isMethod ? SemanticTokenTypes.method : SemanticTokenTypes.function;
                 }
-                if (effectiveType && isOverloaded(effectiveType)) {
-                    // check whether any overload has a method declaration
-                    const isMethod = OverloadedType.getOverloads(effectiveType).some(
-                        (fType) => fType.shared.declaration?.isMethod
+                // If `effectiveType` is a function/overloaded type or a union thereof,
+                // highlight as a function/method
+                if (allSubtypes(effectiveType, (t) => isFunction(t) || isOverloaded(t))) {
```

Comment:
```suggestion
                if (allSubtypes(effectiveType, isFunctionOrOverloaded)) {
```

i think this will also prevent you from needing that cast below